### PR TITLE
openssh: add BRs for check section, patch broken tests

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

OpenSSH package tests currently fail due to missing BuildRequires lines and a couple of broken tests. This commit makes the tests pass.

###### Change Log  
- Add useradd and sudo requirements when running check sections
- Fix failing regression tests due to a bad "future date"

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Local build of package with RUN_CHECK=y

### CODE REVIEW

**Comment by thcrain-msft** [comment]
Opting not to address the remainder of spec linting issues.

File: SPECS/openssh/openssh.spec [hunk 1]
```diff
@@ -25,6 +26,10 @@ BuildRequires:  krb5-devel
 BuildRequires:  openssl-devel
 BuildRequires:  pam-devel
 BuildRequires:  systemd
+%if %{with_check}
+BuildRequires:  %{_sbindir}/useradd
```
Comment (PawelWMS) [inline SPECS/openssh/openssh.spec:30]: I'd go with `shadow-utils` instead. That's what we've been doing so far, it's good to be consistent.

File: SPECS/openssh/regress-test-future-cert-fix.patch [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+diff --git a/regress/cert-hostkey.sh b/regress/cert-hostkey.sh
+index 3ce77796..74d5a530 100644
+--- a/regress/cert-hostkey.sh
++++ b/regress/cert-hostkey.sh
+@@ -248,7 +248,7 @@ test_one() {
+ test_one "user-certificate"	failure "-n $HOSTS"
+ test_one "empty principals"	success "-h"
+ test_one "wrong principals"	failure "-h -n foo"
+-test_one "cert not yet valid"	failure "-h -V20200101:20300101"
++test_one "cert not yet valid"	failure "-h -V20300101:20320101"
```
Comment (PawelWMS) [inline SPECS/openssh/regress-test-future-cert-fix.patch:25]: Might as well go for the next millennium to never have to go back to this. Or at least to be sure we won't be around to worry about it.;)

**Review by anphel31 — APPROVED** [review]

File: SPECS/openssh/openssh.spec (hunk 1 above)
Comment (thcrain-msft) [reply SPECS/openssh/openssh.spec]: Fixed, thanks

File: SPECS/openssh/regress-test-future-cert-fix.patch (hunk 2 above)
Comment (thcrain-msft) [reply SPECS/openssh/regress-test-future-cert-fix.patch]: Sure thing :)

I don't know the date of my own death, so I'll just eyeball it as the late 90s.

**Pull request merged**